core/chainio/avs_reader.go

```diff
@@ -150,3 +154,47 @@ func (r *AvsReader) GetOldTaskHash(nBlocksOld uint64, interval uint64) (*[32]byt
 	batchIdentifierHash := *(*[32]byte)(crypto.Keccak256(batchIdentifier))
 	return &batchIdentifierHash, nil
 }
+
+// Returns a pending batch from its merkle root or nil if it doesn't exist
+// Searches the last `BatchFetchBlocksRange` blocks at most
+func (r *AvsReader) GetPendingBatchFromMerkleRoot(merkleRoot [32]byte) (*servicemanager.ContractAlignedLayerServiceManagerNewBatchV3, error) {
```

Comment:
I can imagine this function being useful in the future. I would refactor it and remove the `responded` check and receive the `fromBlock` as a parameter instead of it being hardcoded. Then the aggregator in `waitForTask` should check for `responded` and calculate the `fromBlock`.

Reply:
I think this would be an easy refactor to make when that moment arrives. For now, I think we should keep it as simple as possible.
Besides that, I'm curious. How would you calculate the `fromBlock`? I think we are left with checking all blocks (`fromBlock = 0`) or with a random constant as I used (`fromBlock = currentBlock - 1000`).

Reply:
Yes, you would still have a hardcoded constant, but the function `GetPendingBatchFromMerkleRoot` would take it as a parameter.